src/extensions/insights/insights.cmd.tsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Color, Box, Text } from 'ink';
+import { Command, CLIArgs } from '../cli';
+import { Flags } from '../paper/command';
+import { InsightManager } from './insight-manager';
+import { InsightResult } from './insight';
+
+export default class InsightsCmd implements Command {
+  name: string;
+  description: string;
+  group: string;
+  opts: string[][];
+  insightManager: InsightManager;
+  constructor(insightManager: InsightManager) {
+    this.insightManager = insightManager;
+    this.opts = [['l', 'list', 'list all insights']];
+    this.name = 'insights [...names]';
+    this.description = 'start a dev environment for a workspace or a specific component';
+    this.group = 'development';
+  }
+
+  async render([names]: CLIArgs, { list }: Flags) {
+    if (list) {
+      //return <Color green>There are many insights</Color>
+      const results = this.insightManager.listInsights();
+      const listItems = results.map(insight => (insight += '\n'));
+      return <Color blueBright>{listItems}</Color>;
+    }
+    if (names) {
+      let results: InsightResult[] = [];
+      if (Array.isArray(names)) {
+        results = await this.insightManager.run(names);
+      } else {
+        results = await this.insightManager.run([names]);
+      }
+      return <Color blueBright>{results}</Color>;
+    } else {
+      const results = await this.insightManager.runAll();
+      // console.log('rendering,',results[0].renderedData)
+      // return results[1].renderedData
+      return (
+        <Box key="help" flexDirection="column">
+          {results.map(function(result) {
+            return (
+              <Box key={result.metaData.name}>
+                <Box>
+                  <Text bold underline>
+                    {result.metaData.name}
+                    {'\n'}
+                  </Text>
+                </Box>
+                <Text>{'\n'}</Text>
+                <Box>{result.renderedData}</Box>
+              </Box>
+            );
+          })}
+        </Box>
```

Comment:
I would move this into another function and do something like
```
results = names ? await this.insightManager.run([names]) : await this.insightManager.runAll();
return template(results)
function template(results) : {
<Box key="help" flexDirection="column">
//....
</Box>
}
```

Reply:
done